core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -267,6 +270,7 @@ public void runInContext() {
       // was cancelled.
       removeContextListenerAndCancelDeadlineFuture();
     }
+    channelStats.reportCallStarted();
```

Comment:
this should be up higher, before the listener can be invoked.   Otherwise the number of calls failed might be higher than the number of calls started; a bug.